csr/csr.go

```diff
@@ -281,6 +281,9 @@ func WatchCastAINodeCSRs(ctx context.Context, log logrus.FieldLogger, client kub
 		waitext.Forever,
 		func(ctx context.Context) (bool, error) {
 			w, err = getWatcher(ctx, client)
+			if errors.Is(err, context.Canceled) {
```

Comment:
maybe add context.DeadlineExceed as well?

Reply:
instead of checking type of error we can check context error:
if err!=nil && ctx.Err!=nil { return false,err}

Reply:
I don't know use case for `context.DeadlineExceed` there is for contex.Canceled. 
To be safe I would like to leave it as it is.